### PR TITLE
Removed the iplant user from all Cfg Mgmt

### PR DESCRIPTION
**Pull request by strootman**

### CODE REVIEW

File: docker/javabase/Dockerfile [hunk 1]
```diff
@@ -12,10 +12,10 @@ RUN apk add --update curl \
...
-RUN busybox adduser -s /bin/sh -u 1337 -D -h /home/iplant iplant
+#RUN mkdir -p /home/iplant
+#RUN busybox adduser -s /bin/sh -u 1337 -D -h /home/iplant iplant
 
-USER iplant
-RUN mkdir -p /home/iplant/logs/
-WORKDIR /home/iplant
+#USER iplant
+#RUN mkdir -p /home/iplant/logs/
+#WORKDIR /home/iplant
```
Comment (ianmcorvidae) [inline docker/javabase/Dockerfile]: Delete these lines for real; we can get them from history if we need them back.

File: services/NotificationAgent/Dockerfile [hunk 2]
```diff
@@ -14,6 +14,5 @@ LABEL org.iplantc.de.notificationagent.git-ref="$git_commit" \
       org.iplantc.de.notificationagent.version="$version" \
       org.iplantc.de.buildenv.git-ref="$buildenv_git_commit"
 
-USER iplant
 ENTRYPOINT ["java", "-Dlogback.configurationFile=/etc/iplant/de/logging/notificationagent-logging.xml", "-cp", ".:notificationagent-standalone.jar", "notification_agent.core"]
```
Comment (ianmcorvidae) [inline services/NotificationAgent/Dockerfile]: This line is depending on things being copied into `/home/iplant` and the workdir being `/home/iplant` -- probably those things will need to be changed to just use `/root` or `/`

Comment (ianmcorvidae) [inline services/NotificationAgent/Dockerfile]: (this of course applies to basically all the service files)

File: ansible/playbooks/grouper.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+        username: Grouper
+        icon: ":fish:"
+        msg: "Deploying Grouper"
+
+      - role: grouper
+
+      - role: notify-chat
+        username: Grouper
+        icon: ":fish:"
+        msg: ":heavy_check_mark: Done deploying Grouper"
```
Comment (psarando) [inline ansible/playbooks/grouper.yaml]: Wasn't this just removed by https://github.com/iPlantCollaborativeOpenSource/DE/pull/12?

**Comment by ianmcorvidae** [comment]
It won't let me comment on the exact lines but you should remove all the things chowning things to the iplant user when you change where files are copied and run from, etc.

File: ansible/roles/de-services-common/tasks/main.yaml [hunk 4]
```diff
@@ -1,27 +1,11 @@
...
-  user: name=iplant shell=/bin/bash groups=iplant append=yes
-  tags:
-      - init_users
-      - init_iplant_user
-
-- name: add the docker group to iplant user
+- name: add the current user to the docker group
   sudo: yes
-  user: name=iplant groups=docker append=yes
+  user: name={{ ansible_ssh_user }} groups=docker append=yes
```
Comment (ianmcorvidae) [inline ansible/roles/de-services-common/tasks/main.yaml]: does this make sense? if I'm reading this right it'll add the person running ansible to the docker group, which should either be something they manage themselves or something done automatically by LDAP or such; it certainly doesn't affect our deployment process

File: services/condor-log-monitor/Dockerfile [hunk 5]
```diff
@@ -9,9 +9,6 @@ LABEL org.iplantc.de.condor-log-monitor.git-ref="$git_commit" \
 
 RUN useradd -m -U -s /bin/bash iplant
```
Comment (ianmcorvidae) [inline services/condor-log-monitor/Dockerfile]: This is part of the same process. But I also think clm might just be in the jex-events image now, so... not sure if it matters? @johnworth would know I guess

File: services/NotificationAgent/Dockerfile (hunk 2 above)
Comment (strootman) [inline services/NotificationAgent/Dockerfile]: Yes, the ansible tasks have been updated to apply owner/group as `root`

File: ansible/playbooks/grouper.yaml (hunk 3 above)
Comment (strootman) [inline ansible/playbooks/grouper.yaml]: Yes. will push a change

File: ansible/roles/de-services-common/tasks/main.yaml (hunk 4 above)
Comment (strootman) [inline ansible/roles/de-services-common/tasks/main.yaml]: You are reading it right, and boy is it convenient!!
And no, it isn't directly needed by the deployment. I'll remove it.

**Comment by strootman** [comment]
@ianmcorvidae I just pushed a changed that removed the chowns

File: ansible/roles/grouper/templates/iplant-grouper.service [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+ExecStop=/usr/bin/docker stop {{grouper.container_name}}
+Restart=on-failure
+StartLimitInterval=60s
+StartLimitBurst=3
+
+SyslogIdentifier={{grouper.container_name}}
+
+[Install]
+WantedBy={{systemd.target}}
+
```
Comment (psarando) [inline ansible/roles/grouper/templates/iplant-grouper.service]: These elk and grouper files, removed in https://github.com/iPlantCollaborativeOpenSource/DE/pull/12, are also getting re-added in this pull request.

Comment (strootman) [inline ansible/roles/grouper/templates/iplant-grouper.service]: Files have been removed. Rebasing onto dev helped identify what was needed to be removed.

File: services/NotificationAgent/Dockerfile [hunk 7]
```diff
@@ -5,7 +5,6 @@ VOLUME ["/etc/iplant/de"]
 
 COPY conf/main/logback.xml /home/iplant/
 COPY target/notificationagent-standalone.jar /home/iplant/
```
Comment (ianmcorvidae) [inline services/NotificationAgent/Dockerfile]: The entrypoint below, in conjunction with these lines' copying locations, still depends on the `WORKDIR /home/iplant` removed from `javabase` (it adds `.` to the classpath, which is the `/home/iplant` here by way of the `WORKDIR`). They should be copied into whatever directory is the default workdir (`/`, I believe, for `jeanblanchard/java:8` which is the base of `javabase`) or another alternately configured working directory such as `/root`.

File: services/jex-events/Dockerfile [hunk 8]
```diff
@@ -10,9 +10,6 @@ LABEL org.iplantc.de.jex-events.git-ref="$git_commit" \
       org.iplantc.de.jex-events.version="$version" \
       org.iplantc.de.buildenv.git-ref="$buildenv_git_commit"
 
-USER iplant
```
Comment (ianmcorvidae) [inline services/jex-events/Dockerfile]: There is a `useradd` command further up in this dockerfile which should be removed.

**Comment by strootman** [comment]
I've removed the `useradd`s, as well as updated the paths for files which are being copied in.

@ianmcorvidae If you find something visually, excellent, but I'm going to start testing these changes out. From building the images to deploying to an environment. I'll keep this PR updated with progress.

**Comment by ianmcorvidae** [comment]
I don't see anything else, checking quickly.

**Pull request merged**